src/library_syscall.js

```diff
@@ -114,10 +114,12 @@ var SyscallsLibrary = {
       }
       return 0;
     },
-    doDup: function(path, flags, suggestFD) {
+    doDup: function(stream, suggestFD) {
       var suggest = FS.getStream(suggestFD);
       if (suggest) FS.close(suggest);
-      return FS.open(path, flags, 0, suggestFD, suggestFD).fd;
+
+      FS.streams[suggestFD] = stream;
```

Comment:
@kripken In future, please ask me about deep POSIX/threading/syscalls/locking/IO/signals an so on questions, I have deep experience in that. You may ask directly via socketpair@gmail.com. Any other developers are welcome.